src/generic/stage2/allocate.h

```diff
@@ -8,9 +8,9 @@ namespace allocate {
 //
 really_inline error_code set_max_depth(dom_parser_implementation &parser, size_t max_depth) {
   parser.containing_scope.reset(new (std::nothrow) scope_descriptor[max_depth]);
-  parser.ret_address.reset(new (std::nothrow) ret_address_t[max_depth]);
+  parser.is_array.reset(new (std::nothrow) bool[max_depth]);
```

Comment:
One nice benefit of this change is that `bool` typically uses one one byte unlike the addresses which would use 8 bytes. It would be possible to use just one bit but at the expense of some work (vector<bool> might implement this 1-bit trade-off).

Reply:
Yeah. In fact, 99.9% of json files could be handled with a single 64-bit integer. (Honestly 32 would be more than enough.) I wasn't quite ready to go that crazy, but it wouldn't be terribly difficult, I think.